backend/app/tests/crud/test_tag_crud.py

```
@@ -0,0 +1,163 @@
+    tag_in_data = {"name": "Test Tag Fail"}
+    tag_in_schema = MockTagCreateSchema(**tag_in_data)
+    MockTagInCrud.return_value = MagicMock()
+    mock_db_session.commit.side_effect = IntegrityError("mock_commit_tag", {}, Exception())
+
+    with pytest.raises(IntegrityError):
+      crud.create_tag(db=mock_db_session, tag_in=tag_in_schema)
+    
+    mock_db_session.rollback.assert_called_once() # crud.create_tag handles rollback
+
+
+@patch("app.crud.Tag") 
```

Ruff (0.11.9)

Trailing whitespace

Remove trailing whitespace

(W291)